build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { MonoRepo, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import { ResetMode } from "simple-git";
+
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import {
+	BaseCommand,
+	type Context,
+	Repository,
+	getPreReleaseDependencies,
+} from "../../library/index.js";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: MonoRepo | Package,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+async function checkNoLocalChanges(context: Context): Promise<CheckResult> {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * Returns true if all the dependencies are installed for the package/release group.
```

Comment:
doc implies this returns a boolean. It does not.

Also would be nice to indicate somehow if this is intended to be used as a CheckFunction. Assuming CheckFunction was well documented, doing so would reduce the need for docs here.